contracts/src/v0.8/dev/functions/AuthorizedOriginReceiver.sol

```diff
@@ -101,7 +101,7 @@ abstract contract AuthorizedOriginReceiver {
           if (s_authorizedSendersList[j] == senders[i]) {
             address last = s_authorizedSendersList[s_authorizedSendersList.length - 1];
             // Copy last element and overwrite senders[i] to be deleted with it
-            s_authorizedSendersList[i] = last;
+            s_authorizedSendersList[j] = last;
             s_authorizedSendersList.pop();
```

Comment:
Morgan also suggested that you could break out of the loop here.

Reply:
Right! Adding.